pkg/scheduler/api/podgroup_info/subgroup_info/factory.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+package subgroup_info
+
+import (
+	"fmt"
+
+	"github.com/NVIDIA/KAI-scheduler/pkg/apis/scheduling/v2alpha2"
+)
+
+const RootSubGroupSetName = ""
+
+func FromPodGroup(podGroup *v2alpha2.PodGroup) (*SubGroupSet, error) {
+	allSubGroups := map[string]*v2alpha2.SubGroup{}
+	children := map[string][]string{}
+
+	for _, subGroup := range podGroup.Spec.SubGroups {
+		if _, found := allSubGroups[subGroup.Name]; found {
+			return nil, fmt.Errorf("subgroup <%s> already exists", subGroup.Name)
+		}
+		allSubGroups[subGroup.Name] = &subGroup
+		parentName := formatParentName(subGroup.Parent)
+		children[parentName] = append(children[parentName], subGroup.Name)
+	}
+
+	root := NewSubGroupSet(fmt.Sprintf("%s/%s", podGroup.Namespace, podGroup.Name), nil)
+	subGroupSets := map[string]*SubGroupSet{
+		RootSubGroupSetName: root,
```

Comment:
Thats a bit something unexpected, 
You have a map that the keys are the name and the values are items which have name.
Most of the time the key equal the value.name, but only in root not.